axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchGenerateMoves.java

```diff
@@ -0,0 +1,435 @@
+      paymentMap =
+          invoiceToolService.getPaymentMap(paymentMap, invoice.getInvoicePaymentList(), invoice);
+    }
+
+    generateMove(
+        journal,
+        account,
+        company,
+        partner,
+        currency,
+        date,
+        period,
+        tradingName,
+        invoiceListByCurrency,
+        accountConfig,
+        amountMap,
+        taxMap,
+        paymentMap);
+  }
+
+  protected void generateMove(
+      Journal journal,
+      Account account,
+      Company company,
+      Partner partner,
+      Currency currency,
+      LocalDate date,
+      Period period,
+      TradingName tradingName,
+      List<Invoice> invoiceList,
+      AccountConfig accountConfig,
+      Map<Account, BigDecimal> amountMap,
+      Map<Tax, BigDecimal> taxMap,
+      Map<PaymentMode, BigDecimal> paymentMap)
+      throws AxelorException {
+
+    Move move =
+        moveCreateService.createMove(
+            journal,
+            company,
+            partner,
+            currency,
+            period,
+            date,
+            tradingName,
+            null,
+            MoveRepository.FUNCTIONAL_ORIGIN_PURCHASE,
+            null,
+            journal.getDescriptionModel());
+
+    int ref = 1;
+    BigDecimal total = BigDecimal.ZERO;
+
+    for (Entry<Account, BigDecimal> entry : amountMap.entrySet()) {
+      ref = generateAccountMoveLine(date, move, ref, entry);
+      total = total.add(entry.getValue());
+    }
+
+    for (Entry<Tax, BigDecimal> entry : taxMap.entrySet()) {
+      ref = generateTaxMoveLine(company, date, move, ref, entry);
+      total = total.add(entry.getValue());
+    }
+
+    MoveLine debitMoveLine = generateDebitMoveLine(account, date, move, ref, total);
+
+    moveRepository.save(move);
+
+    Map<PaymentMode, Move> paymentMoveMap = new HashMap<>();
+
+    for (Entry<PaymentMode, BigDecimal> entry : paymentMap.entrySet()) {
+      Move paymentMove =
+          generatePaymentMove(
+              tradingName,
+              period,
+              company,
+              partner,
+              currency,
+              account,
+              date,
+              move,
+              debitMoveLine,
+              entry);
+      paymentMoveMap.put(entry.getKey(), paymentMove);
+    }
+
+    for (Invoice invoice : invoiceList) {
+      ventilateInvoice(move, invoice, accountConfig);
+      generateInvoicePayment(date, paymentMoveMap, invoice);
+    }
+
+    String origin =
+        invoiceList.stream().map(Invoice::getInvoiceId).collect(Collectors.toList()).toString();
+
+    move.setOrigin(origin);
+
+    Boolean dayBookMode =
+        accountConfigService.getAccountConfig(move.getCompany()).getAccountingDaybook()
+            && move.getJournal().getAllowAccountingDaybook();
+
+    moveValidateService.updateValidateStatus(move, dayBookMode);
+  }
+
+  @Transactional
+  protected InvoicePayment generateInvoicePayment(
```

Comment:
We already have a class to create a new InvoicePayment

https://github.com/axelor/axelor-open-suite/blob/master/axelor-account/src/main/java/com/axelor/apps/account/service/payment/invoice/payment/InvoicePaymentCreateServiceImpl.java#L79